demos/cloud-shell-vnet/main.bicep

```diff
@@ -67,10 +71,46 @@ resource containerSubnet 'Microsoft.Network/virtualNetworks/subnets@2021-08-01'
         }
       }
     ]
+    networkSecurityGroup: {
+      id: nsgResourceId
```

Comment:
```suggestion
      id: networkSecurityGroup.id
```

Reply:
This is variable created in variables area (line 47) and used it. If I replace as suggested, Bicep is throwing errors. I think keeping it like this is simple and reusable.